Find a functionally equivalent code:android.content.res.Resources.obtainTypedArray:COMMENT
Method Modifier: public      
Comment:/**
 * Return an array of heterogeneous values.
 *
 * @param id The desired resource identifier, as generated by the aapt
 * tool. This integer encodes the package, type, and resource
 * entry. The value 0 is an invalid identifier.
 *
 * @throws NotFoundException Throws NotFoundException if the given ID does not exist.
 *
 * @return Returns a TypedArray holding an array of the array values.
 * Be sure to call {@link TypedArray#recycle() TypedArray.recycle()}
 * when done with it.
 */

@@ -3,9 +3,9 @@
     if (len < 0) {
         throw new NotFoundException("Array resource ID #0x" + Integer.toHexString(id));
     }
-    TypedArray array = getCachedStyledAttributes(len);
+    TypedArray array = TypedArray.obtain(this, len);
     array.mLength = mAssets.retrieveArray(id, array.mData);
     array.mIndices[0] = 0;
     return array;

